lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -247,38 +269,40 @@ def add_test_app_host_targets
                 create_info_plist_file(app_host_info_plist_path, app_host_target, '1.0.0', target.platform, :appl)
               end
               # Wire all test native targets with the app host.
-              native_test_target = target.native_target_for_spec(test_spec)
-              native_test_target.build_configurations.each do |configuration|
+              test_native_target = test_native_targets.find do |native_target|
+                native_target.symbol_type == target.product_type_for_test_type(spec_consumer.test_type)
+              end
+              test_native_target.build_configurations.each do |configuration|
                 test_host = "$(BUILT_PRODUCTS_DIR)/#{name}.app/"
                 test_host << 'Contents/MacOS/' if target.platform == :osx
                 test_host << name.to_s
                 configuration.build_settings['TEST_HOST'] = test_host
               end
               target_attributes = project.root_object.attributes['TargetAttributes'] || {}
-              target_attributes[native_test_target.uuid.to_s] = { 'TestTargetID' => app_host_target.uuid.to_s }
+              target_attributes[test_native_target.uuid.to_s] = { 'TestTargetID' => app_host_target.uuid.to_s }
               project.root_object.attributes['TargetAttributes'] = target_attributes
-            end
+            end.compact
```

Comment:
have to add `compact` because we now use `map` and line above includes `next unless...`